### PR TITLE
Initializing t0 in the dependencies list.

### PR DESCRIPTION
**Pull request by vytjan**
Adding the Tier-0 as another WMAgent system to the dependencies list.

### CODE REVIEW

**Comment by amaltaro** [comment]
@vytjan as we discussed today, I'm probably cutting a new release tonight. If you want this one in, please a) fix the conflicts and b) test it once again. Thanks

**Comment by vytjan** [comment]
@amaltaro Alan, this is tested and a new T0 release can be built this way as a WMCore subsystem.
Or is there something missing?

File: setup_dependencies.py [hunk 1]
```diff
@@ -170,5 +170,36 @@
...
+                     'Controllers+', 'WMQuality.Emulators+',
+                     'WMCore.ACDC', 'Utils'],
+        'modules': ['WMCore.Configuration',
+                    'WMCore.DAOFactory',
+                    'WMCore.WMException',
+                    'WMCore.Lexicon',
+                    'WMCore.WMBS.File'],
+        'systems': ['wmc-web', 'wmc-database', 'wmc-runtime', 'workqueue'],
+        'statics': ['src/javascript/WMCore/WebTools/Agent',
+                    'src/javascript/WMCore/WebTools/WMBS',
```
Comment (amaltaro) [inline setup_dependencies.py]: These two "WMCore/WebTools/..." is something that I need to test.
Unless I'm mistaken, this corresponds to the old WMBS service monitoring code that was in place for the agents, given that we have deprecated/removed all that - python - code I think we no longer need it. But again, it has to be tested.

**Review by amaltaro — CHANGES_REQUESTED** [review]
Vytas, I made some comments in the code. If you don't want to test those dependency changes, no worries, just do the rebase and that's all.

File: bin/wmagent-mod-config [hunk 2]
```diff
@@ -130,6 +130,8 @@ def modifyConfiguration(config, **args):
     if hasattr(config, "General"):
         config.General.central_logdb_url = args["central_logdb_url"]
         config.General.centralWMStatsURL = args["wmstats_url"]
+        # Both PhEDExInjector and WorkQueueManager use it, so we better make it general
+        config.General.rucioAccount = args["rucio_account"]
```
[the remark this answers is not in the document]
Comment (vytjan) [reply bin/wmagent-mod-config]: MIssed that this is already in master, thanks! Rebased.

File: setup_dependencies.py [hunk 3]
```diff
@@ -170,5 +170,36 @@
...
+        'packages': ['WMCore.Agent+', 'WMCore.Algorithms+',
+                     'WMCore.JobStateMachine', 'WMComponent+',
+                     'WMCore.ThreadPool', 'WMCore.WorkerThreads',
+                     'WMCore.BossAir+', 'WMCore.Credential',
+                     'WMCore.JobSplitting+', 'WMCore.ProcessPool',
+                     'WMCore.Services+', 'WMCore.WMSpec+',
+                     'WMCore.WMBS+', 'WMCore.ResourceControl+',
+                     'WMCore.DataStructs+', 'WMCore.ReqMgr+',
+                     'Controllers+', 'WMQuality.Emulators+',
+                     'WMCore.ACDC', 'Utils'],
```
[the remark this answers is not in the document]
Comment (vytjan) [reply setup_dependencies.py]: Probably T0 does not use it, it's a leftover from the copied wmagent setup.

File: setup_dependencies.py (hunk 1 above)
Comment (vytjan) [reply setup_dependencies.py]: Removed all 3 WebTools statics, testing it in a replay now. If the replay is fine, then we should be good with these changes.

Comment (vytjan) [reply setup_dependencies.py]: A replay testing these went fine and I didn't spot any issues (related to WebTools or any other).
It should be safe to merge this now.

**Comment by vytjan** [comment]
I mIssed that the rucio config was already in master, thanks! Rebased it.
Removed ACDC and all 3 WebTools statics.
A replay testing these changes went fine and I didn't spot any issues (related to WebTools or any other).
It should be safe to merge this now.

**Comment by amaltaro** [comment]
Thanks, Vytas! Welcome to a new era of T0 deployments ;)

**Review by amaltaro — APPROVED** [review]

**Pull request merged**